overlord/devicestate/devicestate_test.go

```diff
@@ -801,6 +803,12 @@ func (s *deviceMgrSuite) TestDoRequestSerialErrorsOnNoHost(c *C) {
 		c.Skip("cannot run test when http proxy is in use, the error pattern is different")
 	}
 
+	// check internet access
+	_, err := net.LookupHost("nowhere.nowhere.test")
+	if err != nil && strings.Index(err.Error(), "connection") != -1 {
```

Comment:
I'm not sure it's possible to easily tell if the error message means that there is no network, or there's no resolv.conf or something else. Prepped a quick test right here https://play.golang.org/p/Lgg5uuiBxqQ On play.golang.org it fails with:

```
2009/11/10 23:00:00 error: Protocol not available
2009/11/10 23:00:00 temp: false
2009/11/10 23:00:00 timeout: false
```

Locally:

```
2018/03/28 08:28:25 error: lookup nowhere.nowhere.test: no such host
2018/03/28 08:28:25 temp: false
2018/03/28 08:28:25 timeout: false
```

In a systemd container with `--private-network` and removed resolv conf:

```
2018/03/28 08:26:01 error: lookup nowhere.nowhere.test: Temporary failure in name resolution
2018/03/28 08:26:01 temp: false
2018/03/28 08:26:01 timeout: false
```

Same container with with `GODEBUG=netdns=go` (to force Go native resolver), notice how it tries to connect to localhost:53:

```
2018/03/28 08:29:54 error: lookup nowhere.nowhere.test on [::1]:53: read udp [::1]:57908->[::1]:53: read: connection refused
2018/03/28 08:29:54 temp: true
2018/03/28 08:29:54 timeout: false
```

Perhaps we should just have an env variable that causes this test to skip.